pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -312,40 +316,77 @@ func (plugin *cniNetworkPlugin) TearDownPod(namespace string, name string, id ku
 	return plugin.deleteFromNetwork(plugin.getDefaultNetwork(), name, namespace, id, netnsPath, nil)
 }
 
-func podDesc(namespace, name string, id kubecontainer.ContainerID) string {
+func PodDesc(namespace, name string, id kubecontainer.ContainerID) string {
 	return fmt.Sprintf("%s_%s/%s", namespace, name, id.ID)
 }
 
+func GetIpFromResult(res *cnicurrent.Result, rt *libcni.RuntimeConf) string {
+	klog.V(4).Infof("res: %s", res.String())
+	rt_json, err := json.Marshal(rt)
+	if err == nil {
+		klog.V(4).Infof("rt: %s", string(rt_json))
+	}
+
+	for _, ip := range res.IPs {
+		if ip.Interface == nil || *ip.Interface > len(res.Interfaces) {
+			continue
+		}
+		intf := res.Interfaces[*ip.Interface]
+		if intf.Sandbox != "" && intf.Name == rt.IfName {
+			// Found our sandbox interface, use the IP
+			return ip.Address.IP.String()
+		}
+	}
+	// Fall back to first un-associated address for compatibility
+	// with older CNI plugins
+	klog.Warningf("There is no ip address associated with network interface : %s", rt.IfName)
```

Comment:
@AlexeyPerevalov If the plugin is CNIVersion 0.1.0 or 0.2.0 the Result won't have any Interfaces when upconverted to cnicurrent.Result. Instead of unconditionally printing the warning, I would only print the warning if `len(res.Interfaces) > 0`